ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutorUtil.java

```diff
@@ -0,0 +1,234 @@
+        .map(DefaultSchemaInjector::new);
+
+    return statements.stream()
+        .map(stmt -> execute(engine, stmt, ksqlConfig, overriddenProperties, schemaInjector))
+        .filter(queryAndSortedSources -> queryAndSortedSources.getSources() != null)
+        .map(
+            queryAndSortedSources -> new PersistentQueryAndSortedSources(
+                (PersistentQueryMetadata) queryAndSortedSources.getExecuteResult().getQuery().get(),
+                queryAndSortedSources.getSources()
+            ))
+        .collect(Collectors.toList());
+  }
+
+
+  @SuppressWarnings({"rawtypes","unchecked"})
+  private static QueryAndSortedSources execute(
+      final KsqlExecutionContext executionContext,
+      final ParsedStatement stmt,
+      final KsqlConfig ksqlConfig,
+      final Map<String, Object> overriddenProperties,
+      final Optional<DefaultSchemaInjector> schemaInjector
+  ) {
+    final PreparedStatement<?> prepared = executionContext.prepare(stmt);
+    final ConfiguredStatement<?> configured = ConfiguredStatement.of(
+        prepared, overriddenProperties, ksqlConfig);
+    final ConfiguredStatement<?> withSchema =
+        schemaInjector
+            .map(injector -> injector.inject(configured))
+            .orElse((ConfiguredStatement) configured);
+    final ExecuteResult executeResult = executionContext.execute(withSchema);
+    if (prepared.getStatement() instanceof CreateAsSelect) {
+      return new QueryAndSortedSources(
+          executeResult,
+          getSortedSources(
+              (CreateAsSelect)prepared.getStatement(),
+              executionContext.getMetaStore()));
+    }
+    return new QueryAndSortedSources(executeResult, null);
+  }
+
+  private static List<DataSource> getSortedSources(
+      final CreateAsSelect createAsSelect,
+      final MetaStore metaStore) {
+    final Relation from = createAsSelect.getQuery().getFrom();
+    if (from instanceof Join) {
+      final Join join = (Join) from;
+      final AliasedRelation left = (AliasedRelation) join.getLeft();
+      final AliasedRelation right = (AliasedRelation) join.getRight();
+      if (metaStore.getSource(left.getRelation().toString()) == null) {
+        throw new KsqlException("Sourde does not exist: " + left.getRelation().toString());
+      }
+      if (metaStore.getSource(right.getRelation().toString()) == null) {
+        throw new KsqlException("Sourde does not exist: " + right.getRelation().toString());
+      }
+      return ImmutableList.of(
+          metaStore.getSource(left.getRelation().toString()),
+          metaStore.getSource(right.getRelation().toString()));
+    } else {
+      final String fromName = ((AliasedRelation) from).getRelation().toString();
+      if (metaStore.getSource(fromName) == null) {
+        throw new KsqlException("Sourde does not exist: " + fromName);
+      }
+      return ImmutableList.of(metaStore.getSource(fromName));
+    }
+  }
+
+
+  static final class QueryAndSortedSources {
```

Comment:
This name is confusing in light of the `PersistentQueryAndSortSources` class below. How about `ExecuteResultAndSortedSources` instead?